include/beman/optional26/optional.hpp

```diff
@@ -488,7 +488,7 @@ inline constexpr optional<T>::optional(optional<U>&& rhs)
     requires detail::enable_from_other<T, U, U&&> && std::is_convertible_v<U&&, T>
 {
     if (rhs.has_value()) {
```

Comment:
https://eel.is/c++draft/optional#ctor-34 means this needs a paper.

Reply:
Yep, and there is https://eel.is/c++draft/optional#assign-25 as well.

Although in a world where only `optional<T>` exists I believe `std::move(*rhs)` and `*std::move(rhs)` do the same thing. So implementers could have used the as-if rule this whole time. Only for `optional<T&>` there might be a semantic difference. Maybe P2988 is a good vehicle for this?